File: Python/Program/Farm.py
from Business import Business
from Crops import Crops
from Animal import Animal

class Farm(Business):
    def __init__(self, owner: str, name: str, address: str, type_: str, crops: list[Crops], animal: list[Animal]):
        super().__init__(owner, name, address)
        self._type = type_
        self._crops = crops      
        self._animal = animal  

    def get_type(self) -> str:
        return self._type

    def set_type(self, type_: str):
        self._type = type_

    def get_crops(self) -> list[Crops]:
        return self._crops

    def set_crops(self, crops: list[Crops]):
        self._crops = crops

    def get_animal(self) -> list[Animal]:
        return self._animal

    def set_animal(self, animal: list[Animal]):
        self._animal = animal

File: C++/Program/Farm.cpp
#pragma once
#include <iostream>
#include <string>
#include <vector>
#include "Business.cpp"
#include "Crops.cpp"
#include "Animal.cpp"

using namespace std;

class Farm : public Business
{
private:
    string type;
    vector<Crops> crops;
    vector<Animal> animals;

public:
    // Constructor
    Farm() {}

    Farm(string owner, string name, string address, string type, vector<Crops> crops, vector<Animal> animals)
        : Business(owner, name, address)
    {
        this->type = type;
        this->crops = crops;
        this->animals = animals;
    }

    // Getter and Setter for type
    string getType()
    {
        return type;
    }

    void setType(string type)
    {
        this->type = type;
    }

    // Getter and Setter for crops
    vector<Crops> getCrops()
    {
        return crops;
    }

    void setCrops(vector<Crops> crops)
    {
        this->crops = crops;
    }

    // Getter and Setter for animals
    vector<Animal> getAnimals()
    {
        return animals;
    }

    void setAnimals(vector<Animal> animals)
    {
        this->animals = animals;
    }
};